test/common/crudtester.go

```diff
@@ -451,7 +470,35 @@ func (c *FederatedTypeCrudTester) versionName(resourceName string) string {
 	return common.PropagatedVersionName(targetKind, resourceName)
 }
 
-func propagatedVersion(version *fedv1a1.PropagatedVersion, clusterName string) string {
+func (c *FederatedTypeCrudTester) getPrimaryClusterName() string {
+	for name, testCluster := range c.testClusters {
+		if testCluster.IsPrimary {
+			return name
+		}
+	}
+	return ""
+}
+
+func (c *FederatedTypeCrudTester) deleteOneNonPrimaryCluster(clusterNames *[]string) {
+	primaryClusterName := c.getPrimaryClusterName()
+
+	for i, name := range *clusterNames {
+		if name == primaryClusterName {
+			continue
+		} else {
+			*clusterNames = append((*clusterNames)[:i], (*clusterNames)[i+1:]...)
+			break
+		}
+	}
+}
+
+func (c *FederatedTypeCrudTester) propagatedVersion(version *fedv1a1.PropagatedVersion, clusterName string) string {
+	// The primary cluster's namespace version is stored as the
```

Comment:
Can you rephrase this?  It doesn't make sense to me as-is.

Reply:
Updated with the below phrasing. Let me know if that is more clear.
```
	// For namespaces, since we do not store the primary cluster's namespace
	// version in PropagatedVersion's ClusterVersions slice, grab it from the
	// TemplateVersion field instead.
```